csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -208,7 +226,7 @@ private IArrowType GetType(TableFieldSchema field, IArrowType type)
             return type;
         }
 
-        static IArrowReader ReadChunk(BigQueryReadClient readClient, string streamName)
+        static IArrowReader? ReadChunk(BigQueryReadClient readClient, string streamName)
```

Comment:
This changes the internal contract slightly because ReadChunk can now result in a null if the stream doesn't have any rows.